packages/snaps-sdk/src/types/methods/set-state.ts

```diff
@@ -0,0 +1,24 @@
+import type { Json } from '@metamask/utils';
+
+/**
+ * The request parameters for the `snap_setState` method.
+ *
+ * @property key - The key of the state to update. If not provided, the entire
+ * state is updated. This may contain Lodash-style path syntax, e.g.,
+ * `a.b.c`, with the exception of array syntax.
+ * @property value - The value to set the state to.
+ * @property encrypted - Whether to use the separate encrypted state, or the
+ * unencrypted state. Defaults to the encrypted state. Encrypted state can only
+ * be used if the extension is unlocked, while unencrypted state can be used
+ * whether the extension is locked or unlocked.
+ */
+export type SetStateParams = {
+  key?: string;
```

Comment:
Wondering we should require key for this new API 🤔

Reply:
Not sure. Not requiring it seems more flexible, and the API isn't a proper key-value store anyway since we allow nested keys. But if you feel strongly about this, I'm happy to change it.

Reply:
Happy to support both, I was just wondering if it would be simpler for us to require it. Also it might nudge devs to use it more and get cleaner code in return 😄